jwt/src/main/java/me/uport/sdk/jwt/JWTTools.kt

```diff
@@ -196,59 +189,99 @@ class JWTTools(
             throw InvalidJWTException("JWT has expired: exp: ${payload.exp}")
         }
 
-        val ddo: DIDDocument = UniversalDID.resolve(payload.iss)
+        val publicKeys = resolveAuthenticator(header.alg, payload.iss)
+
+        if (header.alg == JwtHeader.ES256K_R) {
+            if (verifyRecoverableES256K(publicKeys, signatureBytes, token.substringBeforeLast('.').toByteArray())) return payload
+        }
+        else if (header.alg == JwtHeader.ES256K) {
+            if (verifyES256K(publicKeys, signatureBytes)) return payload
+        }
+
+        when (header.alg) {
+            JwtHeader.ES256K -> if (verifyES256K(publicKeys,
+                            signatureBytes))
+                return payload
+            JwtHeader.ES256K_R -> if (verifyRecoverableES256K(publicKeys,
+                            signatureBytes,
+                            token.substringBeforeLast('.').toByteArray()))
```

Comment:
The signed data will be used by the other method too

Reply:
I need a bit more explanation on this

Reply:
you'll need to pass the message that is signed to both verification methods.

`val message = token.substringBeforeLast('.').toByteArray(utf8)`

Keep in mind that `ecVerify` expects the message hash so for that method you'll also need to do a `val messageHash = message.sha256()`